src/System.Runtime/tests/System/String.cs

```diff
@@ -1176,156 +1152,126 @@ public static void TestOperators()
+    [InlineData("Hello", "l", "", "Heo")]
+    [InlineData("Hello", "l", null, "Heo")]
+    [InlineData("11111", "1", "23", "2323232323")]
+    [InlineData("111111", "111", "23", "2323")]
+    [InlineData("1111111", "111", "23", "23231")]
+    [InlineData("11111111", "111", "23", "232311")]
+    [InlineData("111111111", "111", "23", "232323")]
+    [InlineData("A1B1C1D1E1F", "1", "23", "A23B23C23D23E23F")]
+    [InlineData("Aa1Bbb1Cccc1Ddddd1Eeeeee1Fffffff", "1", "23", "Aa23Bbb23Cccc23Ddddd23Eeeeee23Fffffff")]
+    [InlineData("11111111111111111111111", "1", "11", "1111111111111111111111111111111111111111111111")] //Checks if we handle the max # of matches
+    [InlineData("11111111111111111111111", "1", "", "")] //Checks if we handle the max # of matches
+    [InlineData("abcdefghijkl", "cdef", "12345", "ab12345ghijkl")]
+    public static void TestReplaceString(string text, string oldValue, string newValue, string expected)
+    {
+        String s = text.Replace(oldValue, newValue);
+        Assert.Equal(expected, s);
+    }
 
-        // (Perf test: If nothing is replaced, don't waste an allocation on a new string.)
-        s = "XYZ";
-        s1 = s.Replace("1", "2");
-        Assert.True(Object.ReferenceEquals(s, s1));
+    [Fact]
+    public static void TestReplaceStringInvalid()
+    {
+        Assert.Throws<ArgumentNullException>(() => "Hello".Replace(null, ""));
+        Assert.Throws<ArgumentException>(() => "Hello".Replace("", "l"));
+    }
 
+    [Theory]
+    [InlineData("XYZ", "1", "2", true)]
+    [InlineData("", "1", "2", true)]
+    public static void TestReplaceStringPerf(string text, string oldValue, string newValue, bool expected)
+    {
         // (Perf test: If nothing is replaced, don't waste an allocation on a new string.)
-        s = "";
-        s1 = s.Replace("1", "2");
-        Assert.True(Object.ReferenceEquals(s, s1));
-
-        // Make sure it can handle the maximum possible # of matches.
-        s = "11111111111111111111111";
-        s1 = s.Replace("1", "11");
-        Assert.Equal("1111111111111111111111111111111111111111111111", s1);
-
-        // Make sure it can handle the maximum possible # of matches.
-        s = "11111111111111111111111";
-        s1 = s.Replace("1", "");
-        Assert.Equal("", s1);
-
-        s = "abcdefghijkl";
-        s1 = s.Replace("cdef", "12345");
-        Assert.Equal("ab12345ghijkl", s1);
-
-        // Cannot pass null for "oldValue"
-        s = "Hello";
-        Assert.Throws<ArgumentNullException>(() => s.Replace(null, "l"));
-
-        // Cannot pass empty string for "oldValue"
-        s = "Hello";
-        Assert.Throws<ArgumentException>(() => s.Replace("", "l"));
-
-        // null is a "valid" input for newValue (equivalent to "")
-        s = "Hello";
-        s1 = s.Replace("l", null);
-        Assert.Equal("Heo", s1);
-
-        return;
+        String s = text.Replace(oldValue, newValue);
+        Assert.Equal(expected, Object.ReferenceEquals(text, s));
```

Comment:
All of the input test cases are expecting true.  We can just change this to not have the ```bool expected``` parameter and instead use ```Assert.Same(text, s)```.